File: Haskell/Mensagens.hs
module Mensagens where
import Util
import Data.List
import System.IO

menuPrincipal :: IO ()
menuPrincipal = do
    putStrLn("\n           -----MENU PRINCIPAL-----")
    putStrLn("\nComo deseja prosseguir?")
    putStrLn("[1] Área do Cliente")
    putStrLn("[2] Login como funcionário")
    putStrLn("[3] Login como dono")
    putStrLn("[4] Sair\n")

opcaoInvalida :: IO()
opcaoInvalida = do
     putStrLn("\nError: OPÇÃO INVÁLIDA\n")

mensagemDeSaida :: IO()
mensagemDeSaida = do
    putStrLn("\nHasta La Vista...E volte sempre!")

menuDoCliente :: IO()
menuDoCliente = do
    putStrLn("\n       -----ÁREA DO CLIENTE-----")
    putStrLn("\nOlá, Cliente!")
    putStrLn("\nComo deseja prosseguir?")
    putStrLn("[1] Listar vagas disponíveis")
    putStrLn("[2] Escolher vaga")
    putStrLn("[3] Recomendação de vaga do Estacionamento")
    putStrLn("[4] Assinar contrato de exclusividade")
    putStrLn("[5] Ocupar vaga para clientes com contrato")
    putStrLn("[6] Voltar ao menu principal\n")

usuarioInvalido :: IO()
usuarioInvalido = do
    putStrLn("\nErro: usuário não cadastrado!\n")

usuarioCadastrado :: IO()
usuarioCadastrado = do
    putStrLn("\nErro: usuário já cadastrado!")

usuarioVagaOcupada :: IO()
usuarioVagaOcupada = do
    putStrLn("\nErro: usuário já está ocupando uma vaga!")

cpfInvalido :: IO()
cpfInvalido = do
    putStrLn("\nErro: CPF inválido!")

menuFuncionario :: IO()
menuFuncionario = do
    putStrLn("\n       -----FUNCIONÁRIO-----")
    putStrLn("\nOlá, funcionário!")
    putStrLn("\nComo deseja prosseguir?")
    putStrLn("[1] Exibir lista das vagas disponíveis")
    putStrLn("[2] Escolher vaga para um cliente")
    putStrLn("[3] Exibir clientes cadastrados")
    putStrLn("[4] Excluir cliente do sistema")
    putStrLn("[5] Calcular valor do estacionamento")
    putStrLn("[6] Voltar ao menu principal\n")

menuFinancas :: IO()
menuFinancas = do
    putStrLn("\n       -----ÁREA DE FINANÇAS-----")
    putStrLn("\nComo deseja prosseguir?")
    putStrLn("[1] Visualizar contratos ativos")
    putStrLn("[2] Alterar valor da hora do estacionamento")
    putStrLn("[3] Voltar ao login do dono\n")

cadastrarNome :: IO()
cadastrarNome = do
    putStrLn("\n       -----CADASTRO DE USUÁRIO-----")
    putStr("\nInforme o nome: ")

informeCpf :: IO()
informeCpf = do
    putStr("Informe o CPF: ")

cadastrarPlaca :: IO()
cadastrarPlaca = do
    putStr("Informe a placa do veículo: ")

cadastraHorarioEntrada :: IO()
cadastraHorarioEntrada = do
    putStr("Por fim, o horário de entrada(números): ")

horaDeSaida :: IO()
horaDeSaida = do
    putStr("Hora de saída: ")

exibirContratosAtivos :: IO()
exibirContratosAtivos = do
    putStrLn("       -----CLIENTES COM CONTRATOS-----\n")
    arq <- openFile "arquivos/contratos.txt" ReadMode
    conteudo <- hGetContents arq
    putStrLn conteudo
    hClose arq

exibirListaClientesCadastrados :: IO()
exibirListaClientesCadastrados = do
    putStrLn("       -----CLIENTES CADASTRADOS-----\n")
    arq <- openFile "arquivos/clientes.txt" ReadMode
    conteudo <- hGetContents arq
    putStrLn conteudo
    hClose arq

menuDono :: IO()
menuDono = do
    putStrLn("\n       -----DONO-----")
    putStrLn("\nOi, chefinho!")

    putStrLn("\nComo deseja prosseguir?")
    putStrLn("[1] Cadastrar funcionário")
    putStrLn("[2] Excluir funcionários")
    putStrLn("[3] Gerenciar finanças")
    putStrLn("[4] Visualizar funcionários ativos")
    putStrLn("[5] Visualizar clientes ativos")
    putStrLn("[6] Voltar ao menu principal\n")


exibirListaFuncionariosCadastrados :: IO()
exibirListaFuncionariosCadastrados = do
    putStrLn("       -----FUNCIONARIOS CADASTRADOS-----\n")
    arq <- openFile "arquivos/funcionarios.txt" ReadMode
    conteudo <- hGetContents arq
    putStrLn conteudo
    hClose arq

cpfParaLogin :: IO()
cpfParaLogin = do
    putStr("\nInforme seu CPF para fazer o login: ")

valorPago :: String -> [[String]] -> IO()
valorPago cpf lista2 = do

    putStr("O valor a ser pago em REAIS pelo cliente " ++ Util.getNome cpf lista2 ++ " é: ")

cadastradoEfetuado :: IO()
cadastradoEfetuado = do
    putStr("\nCADASTRADO EFETUADO COM SUCESSO!")

File: Prolog/mensagens.pl
:- include('util.pl').

menuPrincipal:-
    writeln("\n           -----MENU PRINCIPAL-----"),
    writeln("\nComo deseja prosseguir?"),
    writeln("[1] Area do Cliente"),
    writeln("[2] Login como funcionario"),
    writeln("[3] Login como dono"),
    writeln("[4] Sair\n").

opcaoInvalida:-
    writeln("\nError: OPCAO INVALIDA").

mensagemDeSaida:-
    writeln("\nHasta La Vista...E volte sempre!").

menuDoCliente:-
    writeln("\n       -----AREA DO CLIENTE-----"),
    writeln("\nOla, Cliente!"),
    writeln("\nComo deseja prosseguir?"),
    writeln("[1] Listar vagas disponiveis"),
    writeln("[2] Escolher vaga"),
    writeln("[3] Recomendacao de vaga do Estacionamento"),
    writeln("[4] Assinar contrato de exclusividade"),
    writeln("[5] Ocupar vaga para clientes com contrato"),
    writeln("[6] Voltar ao menu principal").
    
    
usuarioInvalido:-
    writeln("\nErro: usuario nao cadastrado!\n").

usuarioCadastrado:-
    writeln("\nUsuario ja cadastrado!").

usuarioVagaOcupada:-
    writeln("\nErro: usuario ja esta ocupando uma vaga!").

cpfInvalido:-
    writeln("\nErro: CPF invalido!").

menuFuncionario:-
    writeln("\n       -----FUNCIONARIO-----"),
    writeln("\nOla, funcionario!"),
    writeln("\nComo deseja prosseguir?"),
    
    writeln("[1] Listar vagas disponiveis"),
    writeln("[2] Escolher vaga para um cliente"),
    writeln("[3] Exibir clientes cadastrados"),
    writeln("[4] Excluir cliente do sistema"),
    writeln("[5] Calcular valor do estacionamento"),
    writeln("[6] Voltar ao menu principal\n").
    

menuFinancas:-
    writeln("\n       -----AREA DE FINANCAS-----"),
    writeln("\nComo deseja prosseguir?"),

    writeln("[1] Visualizar contratos ativos"), 
    writeln("[2] Alterar valor da hora do estacionamento"), 
    writeln("[3] Voltar ao login do dono\n").
    
cadastrarNome:-
    writeln("\n       -----CADASTRO DE USUARIO-----"),
    writeln("\nInforme o nome: ").

informeCpf:-
    writeln("Informe o CPF: ").

informeNome:-
    writeln("Informe o nome: ").

cadastrarPlaca:-
    writeln("Informe a placa do veiculo: ").

cadastraHorarioEntrada:-
    writeln("Hora de entrada(numeros): ").

horaDeSaida:-
    writeln("Hora de saida: ").

menuDono:-
    writeln("\n       -----DONO-----"),
    writeln("\nOi, chefinho!"),
    writeln("\nComo deseja prosseguir?"),

    writeln("[1] Cadastrar funcionario"),
    writeln("[2] Excluir funcionarios"),
    writeln("[3] Gerenciar financas"),
    writeln("[4] Visualizar funcionarios ativos"),
    writeln("[5] Visualizar clientes ativos"),
    writeln("[6] Voltar ao menu principal\n").


exibirListaFuncionariosCadastrados:-
    writeln("       -----FUNCIONARIOS CADASTRADOS-----"),
    lerArquivoCsv('funcionarios.csv', Lists),
    writeln(Lists).

cpfParaLogin:-
    writeln("\nInforme seu CPF para fazer o login: ").

cadastradoEfetuado:-
    writeln("\nCADASTRADO EFETUADO COM SUCESSO!").
    
msgAssinarContrato:-
    writeln("\nO nosso contrato mensal custa 150$ reais."),
    writeln("Ja o semanal custa 25$ reais."),
    writeln("Qual contrato o senhor ira assinar? [S/M]").